iree/hal/cuda/graph_command_buffer.c

```diff
@@ -140,7 +151,15 @@ static void* iree_hal_cuda_graph_command_buffer_dyn_cast(
 
 static iree_status_t iree_hal_cuda_graph_command_buffer_begin(
     iree_hal_command_buffer_t* base_command_buffer) {
-  // Nothing to do.
+  iree_hal_cuda_graph_command_buffer_t* command_buffer =
+      iree_hal_cuda_graph_command_buffer_cast(base_command_buffer);
+  // TODO(thomasroux): reset existing state, if present. Right now this leaks.
+  if (command_buffer->graph) {
+    return iree_make_status(
+        IREE_STATUS_UNIMPLEMENTED,
+        "rerecording of command buffers not yet supported with CUDA graphs");
+  }
+  iree_hal_resource_set_reset(command_buffer->resource_set);
   return iree_ok_status();
```

Comment:
fyi @ThomasRaoux new TODO with your name on it :P

Reply:
spotted the bug and at least made it bail nicely :)
I've been toying with the idea of making command buffers immutable, so we may not need to fix this: https://github.com/google/iree/issues/8000

Reply:
There should always be a graph as it is created during command_buffer_create. I'm not sure I understand the case it is trying to catch? Can we have this function called several times for the same command buffer? 
I can dig into that more at some point

Reply:
my logic here was wrong, but also what was in there has some bugs: graph is released in end(), so on the second call to begin() it's null. fixing it up (until #8000 lets us simplify)

Reply:
huh that was very wrong indeed. Thanks for fixing it and sorry about that. Let me know if you want me to sort some of this out before the command buffers are made immutable.